metrics-facade-base/src/main/java/com/ringcentral/platform/metrics/instrument/executors/AbstractMonitoredExecutorServiceBuilder.java

```diff
@@ -0,0 +1,176 @@
+        this.parent = parent;
+        this.registry = registry;
+    }
+
+    /**
+     * Sets the name of the executor service.
+     * The name will either be included as part of the metric names (when {@code nameAsLabel(false)} is used),
+     * or as the value of the "name" label (the default behavior with {@code nameAsLabel(true)}).
+     *
+     * @param name the name to assign to the executor service
+     * @return this builder
+     */
+    public B name(String name) {
+        this.name = name;
+        return builder();
+    }
+
+    /**
+     * Configures whether the name of the executor service should be used as the value of the "name" label
+     * or included directly as part of the metric names.
+     *
+     * @param nameAsLabel {@code true} to use the name as a label, {@code false} to include it as part of the metric names
+     * @return this builder
+     */
+    public B nameAsLabel(boolean nameAsLabel) {
+        this.nameAsLabel = nameAsLabel;
+        return builder();
+    }
+
+    /**
+     * Sets the metric name prefix.
+     *
+     * @param metricNamePrefix the prefix to use for the metric names
+     * @return this builder
+     * @throws IllegalArgumentException if {@code metricNamePrefix} is empty
+     */
+    public B metricNamePrefix(@Nonnull MetricName metricNamePrefix) {
+        checkArgument(!metricNamePrefix.isEmpty(), "metricNamePrefix is empty");
+        this.metricNamePrefix = metricNamePrefix;
+        return builder();
+    }
+
+    /**
+     * Sets the label values to prefix the metrics with.
+     *
+     * @param prefixLabelValues the label values to prefix the metrics with
+     * @return this builder
+     */
+    public B prefixLabelValues(LabelValues prefixLabelValues) {
+        this.prefixLabelValues = prefixLabelValues;
+        return builder();
+    }
+
+    /**
+     * Sets additional label values to be applied to the metrics.
+     *
+     * @param additionalLabelValues additional label values to be applied
+     * @return this builder
+     */
+    public B additionalLabelValues(@Nonnull LabelValues additionalLabelValues) {
+        this.additionalLabelValues = requireNonNull(additionalLabelValues);
+        return builder();
+    }
+
+    /**
+     * Helper method to cast the current builder to the concrete builder type.
+     *
+     * @return the concrete builder type
+     */
+    @SuppressWarnings("unchecked")
+    protected B builder() {
+        return (B)this;
+    }
+
+    /**
+     * Builds and returns the monitored {@link ExecutorService}.
+     *
+     * <p>This method constructs the monitored executor service with the provided configuration and registers
+     * metrics using the configured {@link MetricRegistry}.
+     *
+     * @return the constructed {@link ExecutorService}
+     */
+    public ES build() {
+        String resultName = name != null ? name : nextOrderNumAsString();
```

Comment:
I suggest adding some kind of prefix, which will reflect the name of the class, for instance, 

```
String resultName = name != null ? name : parent.getClass().getSimpleName() + "_" + nextOrderNumAsString();
```

It will allow to differentiate between different classes of executors at least.

Reply:
I would suggest adding it as an optional label. What do you think? For example: builder.withExecutorClassLabel(true/false), with true as the default.

Reply:
I will set it to false by default because:
- In most cases, it can be differentiated by the executor name.
- The class name is not always descriptive, for example, for `Executors.newSingleThreadExecutor()`: `executor_service_submitted_total{class="FinalizableDelegatedExecutorService", name="1"} 6.0.`

Reply:
Supported